jaeger/traces.go

```diff
@@ -6,78 +6,81 @@ import (
 	"net/http"
 	"net/url"
 	"path"
+	"strconv"
 	"time"
 
-	jaegerModels "github.com/jaegertracing/jaeger/model/json"
-
 	"github.com/kiali/kiali/config"
 	"github.com/kiali/kiali/log"
 )
 
-func getTraces(client http.Client, endpoint *url.URL, namespace string, service string, rawQuery string) (traces []jaegerModels.Trace, code int, err error) {
+func getTraces(client http.Client, endpoint *url.URL, namespace string, service string, rawQuery string) (response *JaegerResponse, code int, err error) {
 	code = 0
 	u := endpoint
-	if config.Get().ExternalServices.Tracing.NamespaceSelector {
-		service = service + "." + namespace
+	queryService := service
+	if config.Get().ExternalServices.Tracing.NamespaceSelector && namespace != config.Get().IstioNamespace {
+		queryService = fmt.Sprintf("%s.%s", service, namespace)
 	}
 	u.Path = path.Join(u.Path, "/api/traces")
 	q, err := url.ParseQuery(rawQuery)
 	if err != nil {
 		log.Errorf("Jaeger parse query error: %s", err)
 		return
 	}
-	q.Add("service", service)
+	q.Add("service", queryService)
 	u.RawQuery = q.Encode()
 	return queryTraces(client, u)
 }
 
-func getTraceDetail(client http.Client, endpoint *url.URL, traceID string) (trace []jaegerModels.Trace, code int, err error) {
+func getTraceDetail(client http.Client, endpoint *url.URL, traceID string) (response *JaegerResponse, code int, err error) {
 	u := endpoint
 	u.Path = path.Join(u.Path, "/api/traces/"+traceID)
 	return queryTraces(client, u)
 }
 
-func getErrorTraces(client http.Client, endpoint *url.URL, namespace string, service string) (errorTraces int, err error) {
+func getErrorTraces(client http.Client, endpoint *url.URL, namespace string, service string, interval string) (errorTraces int, err error) {
 	errorTraces = 0
 	err = nil
 	u := endpoint
 	u.Path = path.Join(u.Path, "/api/traces")
-
+	log.Errorf("%s", interval)
```

Comment:
This is going to print always a error in the log.
Perhaps this was left for debugging ?